src/main/java/com/google/devtools/build/lib/analysis/RunfilesSupport.java

```diff
@@ -132,15 +139,18 @@ private static RunfilesSupport create(
       runfilesInputManifest = null;
       runfilesManifest = null;
     }
+    Artifact repoMappingManifest = createRepoMappingManifestAction(ruleContext, owningExecutable);
```

Comment:
Do I understand correctly that the same correctness guarantees will apply to the repo manifest as for the runfiles output manifest (and the runfiles tree), which is "nothing"? (I'm not saying this is a bad thing, I don't think you should be solving the correctness problems of runfiles here, just trying to validate my mental model against reality)

Reply:
I don't quite understand what you mean by "correctness guarantees"; any examples?

Reply:
Erm, sorry for being terse. My understanding is that Bazel doesn't guarantee that if one does

```
bazel build //a:b
rm bazel-bin/src/a/b.runfiles/<some file>
bazel build //a:b
```

the runfiles tree is not rebuilt except if `<some file>` is `MANIFEST`. Do I understand correctly that this would be the same with the repo mapping manifest, i.e. deleting it or changing it would not cause the symlink tree creation action to be re-run?

Reply:
I don't really know runfiles well enough to answer this question, but my guess is that yeah, the repo mapping manifest is just like the output manifest.